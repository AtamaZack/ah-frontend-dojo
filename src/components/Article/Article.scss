.inputPanel {
  float: left;
  width: 100%;
  padding: 0 1.25rem 0 1.25rem;
  height: 40rem;
}

.inputPanel img {
  float: right;
  width: 18%;
  margin-top: 8rem;
}

.inputPanel div {
  float: left;
  width: 90%;
}

.inputForm form {
  width: 100%;
  padding: 0.5rem 0;
}

.inputForm input, .inputForm button, .inputForm textarea {
  width: 100%;
  background: #2D658612;
  border: 0.06rem #2D658612 solid;
  border-radius: 0.6rem;
  padding: 0 0.6rem;
  outline: none;
  margin: 0.4rem 0;
}

.inputForm input {
  height: 2.5rem;
  font-size: 1rem;
}

.inputForm button {
  width: 8.12rem;
  height: 2rem;
  font-size: 0.8rem;
  color: #FFFFFF;
  background: #2D6686;
  border: 0.06rem #2D6686 solid;
  cursor: pointer;
}

.inputForm button:hover {
  opacity: 0.9;
}

.inputForm button:active {
  opacity: 1;
}

#description {
  height: 4rem;
  font-size: 1rem;
}

.inputForm textarea {
  height: 20rem;
  padding: 0.6rem;
  font-size: 0.8rem;
}
