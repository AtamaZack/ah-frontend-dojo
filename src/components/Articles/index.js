// react libraries
import React, { Component } from 'react';

// styles
import './Articles.scss';

// third party libraries
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import PropTypes from 'prop-types';

// thunks (action creators)
import { getAllMyAricles } from 'store/actions/articleActions';

class Articles extends Component {
  componentDidMount() {
    // eslint-disable-next-line react/prop-types
    const { getMyAricles } = this.props;
    getMyAricles();
  }

  render() {
    const { articles } = this.props;
    const articlesList = articles.map(article => (
      <div key={article.id} className="articleCard">
        <p className="articleTitle">{ article.title }</p>
        <p className="articleAuthor">{ article.author }</p>
        <p className="articleBody">{ article.body }</p>
      </div>
    ));
    return (
      <div align="center">
        <div className="mainPage">
          <div className="header">
            <div className="right-header-panel" align="right">
              <div className="authContainer" align="center">
                <li><a href="http://localhost:8080/login">Sign in</a></li>
                <li>|</li>
                <li><a href="http://localhost:8080/signup">Sign up</a></li>
              </div>
              <button type="button" className="button botton-size-1">Home</button>
              <button type="button" className="button botton-size-1 active-botton">Articles</button>
            </div>
            <img src="../src/assets/logo.png" alt="logo" className="logo" />
            <div className="breadcrumbs" align="left">
              <ul>
                <li>Articles</li>
              </ul>
            </div>
          </div>
          <div>
            <input type="text" className="search-bar" />
          </div>
          <div className="articlesContainer" align="left">
            {articlesList}
          </div>
        </div>
      </div>
    );
  }
}

Articles.propTypes = {
  articles: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    slug: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    body: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    author: PropTypes.string.isRequired,
    publish_status: PropTypes.bool.isRequired,
    createdAt: PropTypes.string.isRequired,
    updatedAt: PropTypes.string.isRequired,
    delete_status: PropTypes.bool.isRequired,
    tagList: PropTypes.arrayOf(
      PropTypes.string,
    ),
    time_to_read: PropTypes.number.isRequired,
    read_stats: PropTypes.shape({
      views: PropTypes.number.isRequired,
      reads: PropTypes.number.isRequired,
    }),
    likeCount: PropTypes.arrayOf(PropTypes.shape({
      likes: PropTypes.number.isRequired,
      dislikes: PropTypes.number.isRequired,
    })),
  })),
};

Articles.defaultProps = {
  articles: [],
};

const mapStateToProps = state => state.articles;

const mapDispatchToProps = dispatch => bindActionCreators({
  getMyAricles: getAllMyAricles,
}, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(Articles);
