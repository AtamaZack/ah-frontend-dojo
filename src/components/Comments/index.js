// react libraries
import React, { Component } from 'react';

// styles
import './Comments.scss';


export default class Comments extends Component {
  render() {
    return (
      <div className="CommentsContainer">
        <div className="commentForm" align="left">
          <form>
            <textarea type="text" placeholder="Write a comment here ...." />
            <button type="submit">Comment</button>
          </form>
        </div>
        {/* <div className="commentsList">
          <div className="commentCard">
            <div className="commentAuthor">
              <p>asf</p>
            </div>
          </div>
        </div> */}
      </div>
    );
  }
}
