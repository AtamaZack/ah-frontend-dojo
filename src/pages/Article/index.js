/* eslint-disable react/jsx-one-expression-per-line */
/* eslint-disable jsx-a11y/img-redundant-alt */
// react libraries
import React, { Component } from 'react';

// styles
import './ArticlePage.scss';

// components
import Navbar from 'components/NavBar';
import Footer from 'components/Footer';
import ArticleCardLarge from '../../components/ArticleCards';

export default class ArticlePage extends Component {
  componentDidMount() {
  }

  fetchArticle = (e, articleSLug) => {
    e.preventDefault();
    // const { title, description, body } = this.state;
    sessionStorage.setItem('articleSlug', articleSLug);
  };

  render() {
    return (
      <div className="canvas">
        <div className="landing">
          <Navbar />
        </div>
        <div className="breadcrumbs" align="left">
          <ul>
            <li><img src="http://localhost:8080/src/assets/images/breadcrumbs.png" alt="breadcrumbs" />/Articles</li>
          </ul>
        </div>
        <div className="searchPanel">
          <div className="filterPanel">
            <p className="filerHeader"><b>FILTERS:</b></p>
            <p className="filterLinks">
              |<a href="/articles/">My Articles</a>|
              <a href="/articles/">Todays Articles</a>|
              <a href="/articles/">Most Popular</a>|
              <a href="/articles/">Most Liked</a>|
              <a href="/articles/">Trending</a>|
            </p>
          </div>
          <form>
            <input type="text" className="search-bar" />
            <button type="submit" className="search-bar">SEARCH</button>
          </form>
        </div>
        <div align="center" className="articlesCanvas">
          <div className="articleContainer">
            <ArticleCardLarge />
          </div>
        </div>
        <Footer />
      </div>
    );
  }
}
