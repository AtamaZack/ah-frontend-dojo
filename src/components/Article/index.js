/* eslint-disable no-unused-vars */
/* eslint-disable react/destructuring-assignment */
import React, { Component } from 'react';
import axios from 'axios';

import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { getArticle, newArticle } from '../../store/actions/articleActions';
import './Article.scss';

export class Article extends Component {
  constructor(props) {
    super(props);
    this.state = {
      // eslint-disable-next-line react/no-unused-state
      article: {
        title: '',
        description: '',
        body: '',
        tagList: [],
      },
    };
  }

  onChange = (e) => {
    this.setState({ [e.target.name]: e.target.value });
  }

  onSubmit = (e) => {
    e.preventDefault();

    const articleData = {
      article: {
        title: this.state.title,
        description: this.state.description,
        body: this.state.body,
        tagList: [],
      },
    };
    this.props.createArticle(articleData);
  }

  render() {
    return (
      // eslint-disable-next-line react/jsx-no-comment-textnodes
      <div align="center">
        <div className="mainPage">
          <div className="header">
            <div className="right-header-panel" align="right">
              <div className="authContainer" align="center">
                <li><a href="http://localhost:8080/login">{sessionStorage.getItem('username')}</a></li>
                <li>|</li>
                <li><a href="http://localhost:8080/signup">Sign out</a></li>
              </div>
              <button type="button" className="button botton-size-1">Home</button>
              <button type="button" className="button botton-size-1 active-botton">Articles</button>
            </div>
            <img src="../src/assets/logo.png" alt="logo" className="logo" />
            <div className="breadcrumbs" align="left">
              <ul>
                <li><a href="http://localhost:8080/articles/"><b>Articles</b></a></li>
                <li> / </li>
                <li>Create Article</li>
              </ul>
            </div>
          </div>
          <div className="inputPanel" align="left">
            <div className="inputForm">
              {/* <div className="loader"></div> */}
              <form onSubmit={this.onSubmit}>
                <input type="text" name="title" placeholder="Title" value={this.setState.title} onChange={this.onChange} />
                <input type="text" id="description" name="description" placeholder="Description" value={this.setState.description} onChange={this.onChange} />
                <textarea type="text" name="body" placeholder="Body" value={this.setState.body} onChange={this.onChange} />
                <button type="submit">Create Article</button>
              </form>
            </div>
          </div>
          <div>
            <input type="text" className="search-bar" />
          </div>
        </div>
      </div>
    );
  }
}

// eslint-disable-next-line react/no-typos
Article.propTypes = {
  createArticle: PropTypes.func.isRequired,
  // eslint-disable-next-line react/require-default-props
  article: PropTypes.shape({
    title: PropTypes.string,
    description: PropTypes.string,
    body: PropTypes.string,
    tagList: PropTypes.array,
  }),
};

const mapDispatchToProps = dispatch => ({
  createArticle(articleData) { dispatch(newArticle(articleData)); }
});

const mapStateToProps = state => ({
  article: state.article
});

export default connect(mapStateToProps, mapDispatchToProps)(Article);
