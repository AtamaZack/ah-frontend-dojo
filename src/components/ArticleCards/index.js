/* eslint-disable react/prop-types */
// react libraries
import React, { Component } from 'react';

// styles
import './ArticleCardLarge.scss';

// components
import { formatDate } from 'utils';
import Comments from '../Comments';

export default class ArticleCardLarge extends Component {
  render() {
    const {
      title,
      body,
      description,
      author,
      updatedAt,
      time_to_read,
    } = this.props;
    return (
      <div className="mainCanvas">
        <div className="cardCanvas">
          <div className="titlePanel">
            <p>{ title }</p>
          </div>
          <div className="imagePanel">
            <img src="http://localhost:8080/src/assets/images/banner.jpg" alt="article img" />
          </div>
          <div className="flexedRow">
            <div className="columnPanel aurthorPanel">
              <div className="authorAvartar">
                <img src="http://localhost:8080/src/assets/images/avartar.jpg" alt="article img" />
              </div>
              <div className="authorName">
                <p align="left">
                  <b>{ author }</b>
                  <br />
                  <button type="submit" className="buttonSmall buttonLarge">
                    View profile
                  </button>
                </p>
              </div>
            </div>
            <div className="columnPanel align-inText-right lightedParagraph">
              <p>
                <b>Date: </b>
                { formatDate(updatedAt) }
              </p>
              <p>
                <b>Read time: </b>
                { time_to_read }
              </p>
            </div>
          </div>
          <div className="bodyPanel">
            <p className="description">{ description }</p>
            <p>{ body }</p>
          </div>
          <div className="additionalDetails inlinePosition">
            <p>
              <b>Average Rating: </b>
              3.9 (2000)
            </p>
            <p>
              <b>Views: </b>
              {/* { read_stats.views } */}
            </p>
            <p>
              <b>Likes: </b>
              {/* { likeCount.likes } */}
            </p>
            <p>
              <b>Dislikes: </b>
              {/* { likeCount.dislikes } */}
            </p>
          </div>
          <div className="additionalDetails inlinePosition">
            <p>
              <b>Tags: </b>
              <button type="submit" className="buttonSmall">Default</button>
              <button type="submit" className="buttonSmall pythonTag">Python</button>
              <button type="submit" className="buttonSmall javascriptTag">JavaScript</button>
              <button type="submit" className="buttonSmall djangoTag">Django 2.1</button>
              <button type="submit" className="buttonSmall reactTag">React</button>
              <button type="submit" className="buttonSmall reduxTag">Redux</button>
            </p>
          </div>
          <div className="splitLine" />
          <div className="commentsPanel">
            <Comments />
          </div>
        </div>
      </div>
    );
  }
}
