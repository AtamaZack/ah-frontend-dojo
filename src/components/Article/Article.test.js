import React from 'react';
import { shallow } from 'enzyme';
import { Article } from 'components/Article/';

// test to check if the Article component matches the snapshot
describe('Article Component', () => {
  const props = {
    createArticle: jest.fn(),
  };
  const component = shallow(<Article {...props} />);
  it('matches the snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
