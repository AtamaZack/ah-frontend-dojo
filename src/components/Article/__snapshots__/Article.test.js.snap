// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article Component matches the snapshot 1`] = `
<div
  align="center"
>
  <div
    className="mainPage"
  >
    <div
      className="header"
    >
      <div
        align="right"
        className="right-header-panel"
      >
        <div
          align="center"
          className="authContainer"
        >
          <li>
            <a
              href="http://localhost:8080/login"
            />
          </li>
          <li>
            |
          </li>
          <li>
            <a
              href="http://localhost:8080/signup"
            >
              Sign out
            </a>
          </li>
        </div>
        <button
          className="button botton-size-1"
          type="button"
        >
          Home
        </button>
        <button
          className="button botton-size-1 active-botton"
          type="button"
        >
          Articles
        </button>
      </div>
      <img
        alt="logo"
        className="logo"
        src="../src/assets/logo.png"
      />
      <div
        align="left"
        className="breadcrumbs"
      >
        <ul>
          <li>
            <a
              href="http://localhost:8080/articles/"
            >
              <b>
                Articles
              </b>
            </a>
          </li>
          <li>
             / 
          </li>
          <li>
            Create Article
          </li>
        </ul>
      </div>
    </div>
    <div
      align="left"
      className="inputPanel"
    >
      <div
        className="inputForm"
      >
        <form
          onSubmit={[Function]}
        >
          <input
            name="title"
            onChange={[Function]}
            placeholder="Title"
            type="text"
          />
          <input
            id="description"
            name="description"
            onChange={[Function]}
            placeholder="Description"
            type="text"
          />
          <textarea
            name="body"
            onChange={[Function]}
            placeholder="Body"
            type="text"
          />
          <button
            type="submit"
          >
            Create Article
          </button>
        </form>
      </div>
    </div>
    <div>
      <input
        className="search-bar"
        type="text"
      />
    </div>
  </div>
</div>
`;
